File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/PatchExpiringFlagsForUserExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class PatchExpiringFlagsForUserExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        config.setApiKey("YOUR_API_KEY");

        var instructions1 = new InstructionUserRequest();
        instructions1.kind(InstructionUserRequest.KindEnum.ADD_EXPIRE_USER_TARGET_DATE);
        instructions1.flagKey("sample-flag-key");
        instructions1.variationId("ce12d345-a1b2-4fb5-a123-ab123d4d5f5d");
        instructions1.value(16534692);
        instructions1.version(1);

        var instructions = new ArrayList<InstructionUserRequest>(List.of (
            instructions1
        ));

        var patchUsersRequest = new PatchUsersRequest();
        patchUsersRequest.comment("optional comment");
        patchUsersRequest.instructions(instructions);

        try
        {
            var response = new UserSettingsApi(config).patchExpiringFlagsForUser(
                "the-project-key", // projectKey
                "the-user-key", // userKey
                "the-environment-key", // environmentKey
                patchUsersRequest
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling UserSettings#patchExpiringFlagsForUser");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/PatchExpiringFlagsForUserExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class PatchExpiringFlagsForUserExample
{
    fun patchExpiringFlagsForUser()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        val instructions1 = InstructionUserRequest(
            kind = InstructionUserRequest.Kind.addExpireUserTargetDate,
            flagKey = "sample-flag-key",
            variationId = "ce12d345-a1b2-4fb5-a123-ab123d4d5f5d",
            value = 16534692,
            version = 1,
        )

        val instructions = arrayListOf<InstructionUserRequest>(
            instructions1,
        )

        val patchUsersRequest = PatchUsersRequest(
            comment = "optional comment",
            instructions = instructions,
        )

        try
        {
            val response = UserSettingsApi().patchExpiringFlagsForUser(
                projectKey = "the-project-key",
                userKey = "the-user-key",
                environmentKey = "the-environment-key",
                patchUsersRequest = patchUsersRequest,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling UserSettingsApi#patchExpiringFlagsForUser")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling UserSettingsApi#patchExpiringFlagsForUser")
            e.printStackTrace()
        }
    }
}
